### PR TITLE
Replace Python test question with Alpine Linux

### PR DESCRIPTION
**Pull request by nwalters512**
This updates the question used by the grading methods tests to use Alpine Linux instead of the full Python grader. This should allow tests to run much more quickly in CI (the image is orders of magnitude smaller) while still exercising the local external grading pipeline. Note that the production external grading pipeline is still not tested here.

While the previous test had the benefit that it actually tested the Python grader itself, this is not the place to do that. We should have proper CI for each grading image, but that'll have to wait for another PR.

### CODE REVIEW

File: apps/prairielearn/src/tests/helperClient.js [hunk 1]
```diff
@@ -128,14 +129,24 @@ module.exports.saveOrGrade = async (instanceQuestionUrl, payload, action, fileDa
     payload.postData = JSON.stringify(payload.postData);
   }
 
+  // Hacky: if this question is using `pl-file-editor` and not `pl-file-upload`,
+  // assume a single file and massage `fileDate` to match the expected format.
```
Comment (jonatanschroeder) [inline apps/prairielearn/src/tests/helperClient.js:133]: Nit
```suggestion
  // assume a single file and massage `fileData` to match the expected format.
```

**Review by eliotwrobson — APPROVED** [review]
Seems reasonable, should set up the full testing in a followup

File: testCourse/questions/externalGrade/alpine/tests/grade.sh [hunk 2]
```diff
@@ -0,0 +1,13 @@
+#! /bin/sh
+
+mkdir -p /grade/results
+
+cat /grade/student/answer.txt
+
+if cmp /grade/student/answer.txt <(echo -n "correct"); then
```
Comment (jonatanschroeder) [inline testCourse/questions/externalGrade/alpine/tests/grade.sh:7]: I know this is mostly for the test suite itself, but if someone is trying this file manually, `cmp` will complain about spaces and blank lines, which shouldn't matter here. So (this is nit), I'd prefer:
```suggestion
if diff -Bwq /grade/student/answer.txt <(echo -n "correct"); then
```

File: apps/prairielearn/src/tests/gradingMethods.test.js [hunk 3]
```diff
@@ -269,16 +269,14 @@ describe('Grading method(s)', function () {
...
-              'href',
-            );
+            $hm1Body('a:contains("HW9.3. External Grading: Alpine Linux smoke test")').attr('href');
           questionsPage = await (await fetch(iqUrl)).text();
           $questionsPage = cheerio.load(questionsPage);
           assert.lengthOf($questionsPage('button[value="grade"]'), 1);
         });
         it('should submit "grade" action', async () => {
           gradeRes = await saveOrGrade(iqUrl, {}, 'grade', [
-            { name: 'fib.py', contents: Buffer.from(fibonacciSolution).toString('base64') },
```
Comment (jonatanschroeder) [inline apps/prairielearn/src/tests/gradingMethods.test.js:281]: I'm surprised lint didn't pick it up, but I believe `fibonacciSolution` is now unused after this change.

**Review by jonatanschroeder — APPROVED** [review]

**Comment by jonatanschroeder** [comment]
One more issue, in https://github.com/PrairieLearn/PrairieLearn/blob/e0af815b105f4abc45245fc31a42f2b455decbf4/apps/prairielearn/src/tests/gradingMethods.test.js#L24
This file doesn't exist anymore.

File: testCourse/questions/externalGrade/alpine/tests/grade.sh (hunk 2 above)
Comment (nwalters512) [reply testCourse/questions/externalGrade/alpine/tests/grade.sh:7]: Fair, I can make that change, though in my testing it was fine (the editor didn't automatically add spaces or trailing newlines or anything).

Comment (nwalters512) [reply testCourse/questions/externalGrade/alpine/tests/grade.sh]: Done in 4291014.

**Pull request merged**